src/rdsn/src/meta/meta_backup_engine.cpp

```diff
@@ -16,74 +16,93 @@
 // under the License.
 
 #include <dsn/dist/fmt_logging.h>
+#include <dsn/utility/fail_point.h>
 #include <dsn/utility/filesystem.h>
 
-#include "common/backup_common.h"
-#include "common/replication_common.h"
-#include "server_state.h"
+#include "meta_backup_engine.h"
 
 namespace dsn {
 namespace replication {
 
-backup_engine::backup_engine(backup_service *service)
-    : _backup_service(service), _block_service(nullptr), _backup_path(""), _is_backup_failed(false)
+meta_backup_engine::meta_backup_engine(meta_service *meta_svc) : _meta_svc(meta_svc) {}
```

Comment:
I'm a bit of confused can it be replaced by the other version `meta_backup_engine(meta_service *meta_svc, bool is_periodic)`?

Reply:
When meta server restart, it will sync backup information from zk, in this case, it won't initialize `is_periodic` by constructor but other function, `meta_backup_engine(meta_service *meta_svc)` will be used in this case.

Reply:
OK, in this case you can pass is_periodic to false, not needed to add another cstor.

Reply:
Sounds reasonable, done